src/components/Facility/FacilityUsers.tsx

```diff
@@ -52,256 +27,44 @@ export default function FacilityUsers(props: any) {
     prefetch: facilityId !== undefined,
   });
 
-  const {
-    data: facilityUserData,
-    refetch: facilityUserFetch,
-    loading: isLoading,
-  } = useQuery(routes.getFacilityUsers, {
-    query: { offset: offset, limit: limit },
-    pathParams: { facility_id: facilityId },
-    prefetch: facilityId !== undefined,
-  });
-
-  const handlePagination = (page: number, limit: number) => {
-    const offset = (page - 1) * limit;
-    setCurrentPage(page);
-    setOffset(offset);
-  };
-
-  const loadFacilities = async (username: string) => {
-    if (isAddFacilityLoading) {
-      return;
-    }
-    const { res, data } = await request(routes.userListFacility, {
-      pathParams: { username: username },
-    });
-    if (res?.ok && data && facilityUserData) {
-      facilityUserData.results = facilityUserData.results.map((user) => {
-        return user.username === username
-          ? {
-              ...user,
-              facilities: data,
-            }
-          : user;
-      });
-    }
-  };
-
-  const hideLinkFacilityModal = () => {
-    setLinkFacility({
-      show: false,
-      username: "",
-    });
-  };
-
-  const handleCancel = () => {
-    setUserData({ show: false, username: "", name: "" });
-  };
-
-  const handleSubmit = async () => {
-    const username = userData.username;
-    await request(routes.deleteUser, {
-      pathParams: { username: username },
-      onResponse: ({ res }) => {
-        if (res?.status === 204) {
-          Notification.Success({
-            msg: t("user_deleted_successfuly"),
-          });
-        }
+  const { data: userListData, loading: userListLoading } = useQuery(
+    routes.getFacilityUsers,
+    {
+      query: {
+        limit: resultsPerPage.toString(),
```

Comment:
`request` used by useQuery will convert to string for you.

```suggestion
        limit: resultsPerPage,
```